packages/lit-html/src/directives/class-map.ts

```diff
@@ -65,41 +65,53 @@ class ClassMapDirective extends Directive {
             .filter((s) => s !== '')
         );
       }
-      for (const name in classInfo) {
-        if (classInfo[name] && !this._staticClasses?.has(name)) {
-          this._previousClasses.add(name);
+      for (const key in classInfo) {
+        // Account for object keys containing multiple class names separated by whitespace.
+        const names = key.split(/\s/).filter((s) => s !== '');
```

Comment:
I used the same `split` and `filter` arguments that are used above when initializing `this._previousClasses`. Perhaps this code should be deduplicated.

I also considered using a greedy quantifier `/\s+/` in the RegExp, but I opted to use the code that was already here.